apollo-router/tests/integration/subscriptions/mod.rs

```diff
@@ -397,6 +408,7 @@ async fn handle_websocket(mut socket: WebSocket, config: SubscriptionServerConfi
                 _ => {}
             }
         }
+        is_closed.store(true, std::sync::atomic::Ordering::Relaxed);
```

Comment:
There is a return statement in the code above which may cause this line not to be hit. Is this OK? 
```
if socket
                                                    .send(axum::extract::ws::Message::text(
                                                        serde_json::to_string(&complete).unwrap(),
                                                    ))
                                                    .await
                                                    .is_err()
                                                {
                                                    return;
                                                }
                                                ```